packages/react-reconciler/src/ReactFiberScheduler.js

```diff
@@ -814,6 +815,14 @@ function renderRoot(
         }
 
         const returnFiber = sourceFiber.return;
+
+        checkForWrongSuspensePriorityInDEV(
```

Comment:
Let's move this check into `throwException` instead of duplicating the brand check etc. I'm moving around things in throwException and I wouldn't have thought to look here for another "thennable" thing that might be related to my change.